src/main/java/catan/domain/model/game/GameBean.java

```diff
@@ -290,6 +300,17 @@ public void setHexes(Set<HexBean> hexes) {
         this.hexes = hexes;
     }
 
+    public Set<HexBean> getHexesWithCurrentDiceValue() {
+        Set<HexBean> hexesWithDiceNumber = new HashSet<HexBean>();
+        for (HexBean hex : this.hexes) {
+            if (getDiceValue() != null && getDiceValue().equals(hex.getDice())) {
```

Comment:
I would invert this check, as according to game logic you have to search for hexesWithDiceNumber only when diceValue not null